runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcDispatcherClient.java

```diff
@@ -62,31 +67,58 @@ public class GrpcDispatcherClient {
   @GuardedBy("this")
   private final Random rand;
 
+  private final WindmillStubFactoryFactory windmillStubFactoryFactory;
+
+  private final AtomicReference<WindmillStubFactory> windmillStubFactory = new AtomicReference<>();
+
+  private final AtomicBoolean useIsolatedChannels = new AtomicBoolean();
+  private final boolean reactToIsolatedChannelsJobSetting;
+
   private GrpcDispatcherClient(
-      WindmillStubFactory windmillStubFactory,
+      DataflowWorkerHarnessOptions options,
+      WindmillStubFactoryFactory windmillStubFactoryFactory,
       DispatcherStubs initialDispatcherStubs,
       Random rand) {
-    this.windmillStubFactory = windmillStubFactory;
+    this.windmillStubFactoryFactory = windmillStubFactoryFactory;
+    if (DataflowRunner.hasExperiment(
+        options, STREAMING_ENGINE_USE_JOB_SETTINGS_FOR_ISOLATED_CHANNELS)) {
```

Comment:
why do we need experiment in addition to the job setting?

Reply:
Discussed, leaving as is and will remove once safe. Defaulting to on will let us cleanup eventually and have released sdks use preferred behavior.